### ISSUE TITLE
ALTCPU-branch fails (no crash, but garbage output) with AXV512 on GCC-8.1

### ISSUE THREAD

**Issue by bforsbe** [3dem/relion#622]
The standard 3D-benchmark classification using the accelerated CPU-instructions set (```--cpu```) produces garbage output at iteration 7, if compiled using gcc 8.1 and the following;
```
cmake  -DCMAKE_C_FLAGS="-march=skylake-avx512 -mtune=skylake-avx512" -DCMAKE_CXX_FLAGS="-march=skylake-avx512 -mtune=skylake-avx512" -DALTCPU=ON $srcDir
```
Compiling and using the accelerated CPU-instructions without avx512-flags or even with the following avx512-flags works as expected:  
```
cmake  -DCMAKE_C_FLAGS="-mavx512f -mavx512cd" -DCMAKE_CXX_FLAGS="-mavx512f -mavx512cd" -DALTCPU=ON $srcDir
```
Using gcc 9.1 and ```-march=skylake-avx512 -mtune=skylake-avx512``` however produces the expected output. This issue is filed to document an incompatibility, which will not be investigated further at the moment.

**Comment by biochem-fan** [comment]
@CharlesCongdon Do you have time to look at this? If this is a GCC bug, we don't have to fix it. I will put "don't use gcc 8.x" in README.

Another suspicious case with GCC 8.3 is reported in https://www.jiscmail.ac.uk/cgi-bin/wa-jisc.exe?A2=ind2011&L=CCPEM&O=D&P=31081.

**Comment by biochem-fan** [comment]
I don't know if related to this but I found a case where GCC 8.3.1 20190507 (Red Hat 8.3.1-4) produced a completely broken binary even without `ALTCPU=ON`. FFTW calls are broken and `relion_image_handler --rescale_angpix` gave a wrong result.

The host CPU is Xeon Gold 6250L (Cascade Lake) so gcc does enable `-march=skylake-avx512` by default (`-march=native`).

Compiling with `-DMKLFFT=ON` or with GCC 10.2.0 solved the issue.

**Comment by biochem-fan** [comment]
FFTW itself is broken.

```
$ cd external/fftw/fftw3
$ tests/bench --verbose=1   --verify 'obr12x3x16x11'`
Found relative error 1.180823e+00 (freq shift)
       0  17.935256958008   0.000000000000    -6.947862625122   0.000000000000
       1  60.251049041748   0.000000000000    85.284652709961   0.000000000000
       2  25.926866531372   0.000000000000     1.299190521240   0.000000000000
       3  16.729934692383   0.000000000000     8.353242874146   0.000000000000
       4  45.980812072754   0.000000000000    43.131580352783   0.000000000000
       5  -8.878332138062   0.000000000000    45.969619750977   0.000000000000
       6 -57.623508453369   0.000000000000    -2.775558471680   0.000000000000
       7 -40.835449218750   0.000000000000   -43.684688568115   0.000000000000
...
```

FFTW release note says:

> FFTW 3.3.8 May 28th, 2018
> Fixed AVX, AVX2 for gcc-8.
>
> By default, FFTW 3.3.7 was broken with gcc-8. AVX and AVX2 code assumed that the compiler honors the distinction between +0 and -0, but gcc-8 -ffast-math does not. The default CFLAGS included -ffast-math . This release ensures that FFTW works with gcc-8 -ffast-math , and removes -ffast-math from the default CFLAGS for good measure.
http://www.fftw.org/release-notes.html

We are using FFTW 3.3.4; we should migrate to 3.3.8.

**Comment by biochem-fan** [comment]
Fixed in b92ca6d.
